src/igvfd/audit/measurement_set.py

```diff
@@ -718,6 +718,30 @@ def audit_missing_external_image_url(value, system):
         )
 
 
+def audit_missing_primer_designs(value, system):
+    '''
+    [
+        {
+            "audit_description": "TAP-seq measurement sets are expected to link to primer designs.",
+            "audit_category": "missing primer designs",
+            "audit_level": "NOT_COMPLIANT"
+        }
+    ]
+    '''
+    preferred_assay_title = value.get('preferred_assay_title')
+    audit_message = get_audit_message(audit_missing_primer_designs, index=0)
+    if preferred_assay_title == 'TAP-seq' and not value.get('primer_designs', ''):
+        detail = (
+            f'Measurement set {audit_link(path_to_text(value["@id"]), value["@id"])} '
+            f'is missing `primer_designs`.'
+        )
```

Comment:
I'm not really interested in auditing the content type of each file linked in primer_designs, this audit will serve to just flag TAP-seq measurement sets without this property.